File: 1497.py
from collections import Counter
from typing import List


class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        f = Counter(n % k for n in arr)

        for key in f.keys():
            other_num = (k - key) % k

            if key == other_num and f[key] % 2 != 0:
                return False

            if f[key] != f[other_num]:
                return False

        return True

File: 1497.cpp
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
  bool canArrange(vector<int> &arr, int k) {
    unordered_map<int, int> f;

    for (int n : arr) {
      int rem = ((n % k) + k) % k;
      f[rem]++;
    }

    for (auto [key, other] : f) {
      int other_key = (k - key) % k;

      if (other_key == key && f[key] % 2 != 0)
        return false;

      if (f[other_key] != f[key])
        return false;
    }

    return true;
  }
};
